### ISSUE TITLE
[UX] [Aladdin] Recommendation doesn't match command

### ISSUE THREAD

**Issue by chenlomis** [Azure/azure-cli#15438]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
Recommending me group create when I'm trying to get credentials for AKS

**To Reproduce**
![image](https://user-images.githubusercontent.com/30876406/95388081-80987300-08a6-11eb-8646-443d881280ba.png)

**Expected behavior**
recommendations is at least AKS related

**Environment summary**

**Additional context**
![image](https://user-images.githubusercontent.com/30876406/95388145-97d76080-08a6-11eb-95b1-2e3f0ab4def4.png)

**Comment by chenlomis** [comment]
+ @christopher-o-toole

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness.

**Comment by yungezz** [comment]
hi @houk-ms could you pls check if this related to error recommendation framework? thanks

**Comment by houk-ms** [comment]
Currently, we don't provide recommendations for resource related validation errors in CLI core.

@chenlomis Are you getting the recommended results from ai-did-you-mean-this extension?

**Comment by chenlomis** [comment]
Hey all
Thanks for all the attention
I'm not entirely sure whether this is part of the did-you-mean-this-extension...IIRC that's already integrated into core is it?  If that's the case then the recommendations are still the same except we extended it the output from 1 recommendation to 3 recommendations for the purpose of the lab

**Comment by yungezz** [comment]
Another case I met the recommendation/error could be more accurate. cc @houk-ms for awareness.
```
catherine@Azure:~$ az extension add databricks
ArgumentParseError: unrecognized arguments: databricks
Try this: 'az extension add --name anextension'
Still stuck? Run 'az --help' to view all commands or go to 'https://docs.microsoft.com/en-us/cli/azure/reference-index?view=azure-cli-latest' to learn more
catherine@Azure:~$ az extension add --name databricks
```

**Comment by houk-ms** [comment]
@chenlomis You're right, while the difference is for resource related validation error, like some resources 'not exist', 'already exist', we don't provide recommedation currently in Core. Therefore, your command will only get the error message in latest CLI version like below.
```
(env_new) PS C:\Users\houk\Desktop\msws\azure-cli> az aks get-credentials -g houk-test -n sample
ResourceNotFoundError: The Resource 'Microsoft.ContainerService/managedClusters/sample' under resource group 'houk-test' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
```

@yungezz I think this is aready a very good recommendation. Surely, it will be better if we can extract the users' params and apply it into the recommended result.

Anyway, let's leave the issue here to share more examples which we think are not accurate/smart enough.

**Comment by houk-ms** [comment]
`az vm delete -n` will give the recommendation `az vm delete --ids @-`, is the `@-` by design?

**Comment by houk-ms** [comment]
`az vm create` has the `{}` wrapped argument `az vm create --admin-password <{admin-password}> --admin-username <deploy> --image <Centos> --name <MyVm> --resource-group <MyResourceGroup>`
